Add trailing slash workaround to generic odata client (#200)
SAP/cloud-sdk: docs/java/features/odata/generic-odata-client.mdx

This is essentially what happens under the hood when you are using the generated type-safe client code.
When we release a support for type-safe calls you can simple replace this snippets with the fluent API.

### Adding Trailing Slash to the Service Path

By default the SAP Cloud SDK strips all trailing slashes from a resource path.
It's a standard behavior and works for 99% of productively available services.

> Sometimes services might complain or even fail if services path doesn't end with a slash `/`.

Reviewer: ```suggestion
Sometimes, however, services might complain or even fail if a path doesn't end with a slash `/`.
```

> - `https://host/servise-path/resource-path/?$filter...` - works

Reviewer: ```suggestion
- `https://host/service-path/resource-path/?$filter...` - works
```

> - `https://host/servise-path/resource-path?$filter..` - doesn't work (default URI composition strategy by the SAP Cloud SDK)

Reviewer: ```suggestion
- `https://host/service-path/resource-path?$filter..` - doesn't work (default URI composition strategy in the SAP Cloud SDK)
```

> To workaround this situation you can use generic OData client

Reviewer: ```suggestion
To workaround this situation you can use generic OData client:
```

> To workaround this situation you can use generic OData client:

Reviewer: ```suggestion
To workaround this situation you can use the generic OData client:
```
